lmdk/cmake/xtensa-toolchain.cmake

```diff
@@ -18,11 +18,11 @@ cmake_path(APPEND XTENSA_TOOLCHAIN_PATH "XtensaTools" OUTPUT_VARIABLE TOOLCHAIN_
 set(CROSS_COMPILE xt-)
 
 # clang or xcc
-find_program(CMAKE_C_COMPILER NAMES ${CROSS_COMPILE}xcc
+find_program(CMAKE_C_COMPILER NAMES ${CROSS_COMPILE}clang
```

Comment:
> How do we align compiler with Zephyr and base FW ?

By leveraging the Zephyr build system and writing CMake code that integrates with it. The current lmdk/CMake code totally ignores it and does everything from scratch, so it's hopeless right now.

Reply:
I don't see any objections against integrating lmdk build system with python build scripts which we use in sof. I thought that is it better to start small and end big?

Reply:
I prefer an iterative approach to make PRs easy to review. We are still working on loadable modules and step by step we are reaching the final solution.

Reply:
Ok, so we should warn users in the Cmakefile that today CC flags and CC name/version are not automatically aligned with the BaseFW/Zephyr kconfig, but its a TODO.  Ultimately we will have a golden Kconfig and west tag for each product that should be easy to align with.

Reply:
> I prefer an iterative approach to make PRs easy to review. We are still working on loadable modules and step by step we are reaching the final solution.

Sounds great as long as the longer term direction is clearly defined, agreed and documented somewhere[*]. At the very very least with some FIXME and TODOs pointing at the longer term so:
- Small, incremental steps never go backwards or in circles. Typically: from "drive-by", ignorant contributors but not just.
- Reviewers are never confusing work in progress with the final, desired solution(s)

> Ok, so we should warn users in the Cmakefile that today CC flags and CC name/version are not automatically aligned with the BaseFW/Zephyr kconfig, but its a TODO.

Perfect example.

[*] which does NOT mean "set in stone".

Reply:
Marc dragged me over to this one too.  FWIW, another option would be to decide on a SDK/ABI/toolchain spec for each hardware variant, and write a little glue code here and in the Zephyr app to enforce it.

It can be as simple as a section of a header that fails to build if conditions aren't met.  Almost all (but surely there are exception) ABI/compatibility-impactful build settings are visible in the preprocessor.   That is "I'm told I need to build and link on a device with this XCHAL_CORE_ID, xt-clang version 76.8.22 or higher, using the CALL0 ABI, yada yada".

Then the problem becomes the simpler one of synchronizing those header-check blobs between the projects (maybe have them both import a single tree via west, etc...).  For extra credit it could contain a hash or something that gets emitted into the object files and you could validate that it matches at load time on the device.

Reply:
So there are some good options here to align FW, modules, CC, tools - I think the 1st step is to put large comments in the Cmake and in the Kconfig as a warning as this unblocks and gets more users of the code who will also be able to contribute to alignment processes.

Reply:
I don't get what are concerns. These modules are build to binary being loaded and being totally independent from FW.

Reply:
> loaded and being totally independent from FW.

@pjdobrowolski can you explain why ABIs don't matter here?

Reply:
I don't say that ABIs don't matter however since this feature is developed by our team only I must say that we have it under control quite good and things like that are important and we keep them in mind. That is why we've extracted headers for modules, we've created separate cmake files and did so many attempts in that matter.

Reply:
> I don't say that ABIs don't matter

Thanks for clarifying.

> since this feature is developed by our team only I must say that we have it under control quite good and things like that are important and we keep them in mind.

I understand but people, teams and assignments come and go while open-source projects tend to last much longer. As you can see here and in #8365 there is also clear interest for this beyond just a single company. ABI issues can cost weeks of work and project delays, so having subtle things "kept in mind" is clearly not enough. The very least and all we've been asking at this early stage is just some comments in the source like "Warning: ABI dragons here. Keep toolchains and flags synchronized". I don't get the resistance; now this discussion took longer to type :-)